### ISSUE TITLE
Error in PCA test after update to SKLL 2.0

### ISSUE THREAD

**Issue by aloukina** [EducationalTestingService/rsmtool#347]
After change to SKLL 2.0 we get the following error:
```
======================================================================
ERROR: test_analyzer.TestAnalyzer.test_compute_pca_less_components_than_features
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/aloukina/opt/miniconda3/envs/rsmtool_skll/lib/python3.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/Users/aloukina/Tools/rsmtool/rsmtool/tests/test_analyzer.py", line 146, in test_compute_pca_less_components_than_features
    (components, variance) = Analyzer.compute_pca(df, df.columns)
  File "/Users/aloukina/Tools/rsmtool/rsmtool/rsmtool/analyzer.py", line 466, in compute_pca
    pca.fit(df_pca)
  File "/Users/aloukina/opt/miniconda3/envs/rsmtool_skll/lib/python3.7/site-packages/sklearn/decomposition/pca.py", line 341, in fit
    self._fit(X)
  File "/Users/aloukina/opt/miniconda3/envs/rsmtool_skll/lib/python3.7/site-packages/sklearn/decomposition/pca.py", line 407, in _fit
    return self._fit_full(X, n_components)
  File "/Users/aloukina/opt/miniconda3/envs/rsmtool_skll/lib/python3.7/site-packages/sklearn/decomposition/pca.py", line 426, in _fit_full
    % (n_components, min(n_samples, n_features)))
ValueError: n_components=101 must be between 0 and min(n_samples, n_features)=100 with svd_solver='full'
```

**Comment by aloukina** [comment]
Part of the issue is due to this change in sklearn: https://github.com/scikit-learn/scikit-learn/pull/8742

The original test has 101 features and 100 samples and the code sets n_components = n_features.
Increasing the size of the data frame resolves this. 

However now we get 101 components instead of 100.

**Comment by desilinguist** [comment]
Addressed by #349.
